projects/angular/src/data/datagrid/datagrid.spec.ts

```diff
@@ -448,13 +448,50 @@ class TabsIntegrationTest {
     </clr-datagrid>
   `,
 })
-class PanelTrackByTest {
+class PanelIteratorTrackByTest {
   items = Array.from(Array(3), (v, i) => {
     return { name: v, id: i };
   });
   trackById: TrackByFunction<{ id: number }> = (index, item) => item.id;
 }
 
+@Component({
+  template: `
+    <clr-datagrid [clrDgItemsTrackBy]="trackById">
+      <clr-dg-column>Item</clr-dg-column>
+      <clr-dg-row *ngFor="let item of items" [clrDgItem]="item">
+        <clr-dg-cell>{{ item.id }}</clr-dg-cell>
+      </clr-dg-row>
+      <clr-dg-detail *clrIfDetail></clr-dg-detail>
+    </clr-datagrid>
+  `,
+})
+class PanelDatagridTrackByTest {
+  items = Array.from(Array(3), (v, i) => {
+    return { name: v, id: i };
+  });
+  trackById: ClrDatagridItemsTrackByFunction<{ id: number }> = item => item.id;
+}
+
+@Component({
+  template: `
+    <clr-datagrid [clrDgItemsTrackBy]="datagridTrackById">
+      <clr-dg-column>Item</clr-dg-column>
+      <clr-dg-row *ngFor="let item of items; trackBy: iteratorTrackBy" [clrDgItem]="item">
+        <clr-dg-cell>{{ item.id }}</clr-dg-cell>
+      </clr-dg-row>
+      <clr-dg-detail *clrIfDetail></clr-dg-detail>
+    </clr-datagrid>
+  `,
+})
+class PanelDatagridAndIteratorTrackByTest {
```

Comment:
Can you add a test case here to match the use case we're addressing with this PR (iteratorTrackBy without datagridTrackBy)?

Reply:
I did. `PanelIteratorTrackByTest`.

Reply:
None of these tests validate the original bug exists, or is fixed. Please add tests asserting that when configured as a multi-select datagrid and a row has ` [clrDgSelectable="false"]`, the checkboxes render in the disabled state when using datagridTrackById and they render in a non-disabled state when using just the iteratorTrackBy

Reply:
Done.

Reply:
Thanks for adding.